### PR TITLE
Proper validation on context sentences

### PR DESCRIPTION
**Pull request by Valseer**
Related Issue #238 

Changes:
- context sentences dynamically check if there is a word or word form in them
-
-

Breaking Changes:
- none known

### CODE REVIEW

File: client/src/components/Lesson/WordCreator/ContextSentences.tsx [hunk 1]
```diff
@@ -2,19 +2,22 @@ import * as React from 'react';
...
 } from 'material-ui';
-
+import{WordInfo, WordForm} from '../../../reducers/lessonReducer';
 import {
     ChangeEvent
 } from '../../util/ChangeEvent';
 
 export interface ContextSentencesProps {
     name: string;
+    word: WordInfo;
```
Comment (mwitte98) [inline client/src/components/Lesson/WordCreator/ContextSentences.tsx]: Update variable name to wordinfo

File: client/src/components/Lesson/WordCreator/ContextSentences.tsx [hunk 2]
```diff
@@ -26,6 +29,10 @@ export class ContextSentences extends React.Component<ContextSentencesProps, Con
 
     constructor(props: ContextSentencesProps) {
         super(props);
+        this.state = {
+            value: ([] as string[]),
+            error: ([] as string[]),
```
Comment (mwitte98) [inline client/src/components/Lesson/WordCreator/ContextSentences.tsx]: Use `error: string[] = []` syntax

File: client/src/components/Lesson/WordCreator/ContextSentences.tsx [hunk 3]
```diff
@@ -39,13 +46,36 @@ export class ContextSentences extends React.Component<ContextSentencesProps, Con
             newVal = newVal.filter((sentence, i) =>
                 sentence !== '' || i === this.props.value.length);
             this.props.onChange(newVal);
+            if (ev.target) {
+                const search = ([] as number[]);
```
Comment (mwitte98) [inline client/src/components/Lesson/WordCreator/ContextSentences.tsx]: Update syntax like above and change variable name. Using `search` and `search()` is a bit confusing.

**Review by mwitte98 — CHANGES_REQUESTED** [review]
A few minor things

File: client/src/components/Lesson/WordCreator/ContextSentences.tsx (hunk 2 above)
Comment (Valseer) [reply client/src/components/Lesson/WordCreator/ContextSentences.tsx]: that form of defining an empty string array doesn't work, this works, if i knew what the issue was with this I could respond better.

Comment (mwitte98) [reply client/src/components/Lesson/WordCreator/ContextSentences.tsx]: Just noticed that we have the type defined above in the interface so I wouldn't include the type here (`error: []`).

We should still do `const result: number[] = [];` below though

**Review by mwitte98 — APPROVED** [review]

**Pull request merged**